doc/sphinx/proofs/writing-proofs/equality.rst

```diff
@@ -39,15 +40,22 @@ Tactics for simple equalities
 
 .. tacn:: reflexivity
 
-   For a goal with the form :n:`{? forall @open_binders , } t = u`,
-   verifies that `t` and `u` are
-   :term:`definitionally equal <definitional equality>`, and if so,
-   solves the goal (by applying `eq_refl`).  If not, it fails.
-
-   The tactic may also be applied to goals with the form
-   :n:`{? forall @open_binders , } R @term__1 @term__2` where
-   `R` is a reflexive relation registered with the `Equivalence` or `Reflexive`
-   typeclasses.  See :cmd:`Class` and :cmd:`Instance`.
+   After doing an :tacn:`intros`,
+   if the resulting goal is in the form `t = u` in which `t` and `u` are
+   :term:`definitionally equal <definitional equality>`, the tactic
+   proves the goal (by applying `eq_refl`).  If not, it fails.
+
+   For example, `n * n = n * n` and `1 = 1 + 2` and `1 + 2 + n = 3 + n` and
+   `0 + n = n` all succeed.
+   However, `n + 0 = n` isn't definitionally equal because `n + 0` is not reducible
+   due to how the add function is defined (see the aside
+   :ref:`here <reversed_add_example>`).
+   `n + 1 + 2 = n + 3` doesn't work because the left hand side is represented
+   as `(n + 1) + 2`, which isn't reducible.
```

Comment:
I fail to see how this adds anything to the paragraph above. This is a reference manual, not an introductory book on the usage of tactics.

This is drowning the reader in irrelevant blabber that furthermore may be contextually false, since there is no reason that the arithmetical operations represented by these symbols are the one from the Coq.Init.Nat module. It may very well be the case that a newbie is using some other implementation of + that happens to compute on the right.

Reply:
> the aside shows that left and right computation may not reduce the same.

This is not my point, I'm talking about the fact that + can mean something else. Even in an innocuous IDE setting, it suffices e.g. to have Zarith imported.

Reply:
It works for me.  There must be more to your example.  (OTOH, how do I say that `n` is an integer?  `Z` is not accepted.  Sadly, it's not easy to discover what it should be from the coqdoc.)

![image](https://github.com/user-attachments/assets/1aa0bf68-8779-4715-b778-eb778644cf5b)

Reply:
How so?  Reduction behaves the same with ZArith:

![image](https://github.com/user-attachments/assets/227fd462-e8e2-4e07-91ea-b6309bc45f4d)

Reply:
This is true, but this is still missing the point. Now + is the Z addition function and it's a coincidence that it computes the same on this precise example. Retry with
```coq
From Coq Require Import Reals.

Open Scope R.

Goal forall x, 0 + x = x.
Proof.
intros.
simpl.
```

Reply:
OK, I revised it to be clearer about the type.  WDYT?

Interestingly, neither `x + 0` nor `1 + 2` are reduced for reals.  I didn't expect that.  (How does one simplify a real expression?)

Reply:
@ppedrot If you have no further comments, perhaps someone in @coq/doc-maintainers would kindly merge?